sos/report/plugins/ubuntu.py

```diff
@@ -22,3 +22,12 @@ def setup(self):
             "hwe-support-status --verbose",
             "ubuntu-advantage status"
         ])
+
```

Comment:
What would you think if we add another check to validate that `ubuntu-advantage-tools `is installed before looking at collecting ubuntu-advantage logs ?

`if self.is_installed('ubuntu-advantage-tools'):`

Reply:
Good idea; I'll get that tested on my side.

Reply:
And move the `ubuntu-advantage status` command under the is_installed() check as well ? 

I'm also starting to think that ubuntu-advantage should probably have its own plugin ? Have you considered that scenario ? so that ubuntu.py will remain community and ua.py will be more ubuntu "entreprise" ish.

Reply:
I can do the first part of that.

For the latter part of that, it isn't clear whether that would be beneficial for sosreport. "ubuntu-advantage-tools" is installed by default on modern community-level/non-enterprise Ubuntu systems, so the additional plugin would be executed along with this plugin. I believe that this is what the output looks like for a non-enterprise, community-level Ubuntu system (UA Tools version 20.3):
```
SERVICE       AVAILABLE  DESCRIPTION
cc-eal        no         Common Criteria EAL2 Provisioning Packages
esm-apps      yes        UA Apps: Extended Security Maintenance
esm-infra     yes        UA Infra: Extended Security Maintenance
fips          no         NIST-certified FIPS modules
fips-updates  no         Uncertified security updates to FIPS modules
livepatch     yes        Canonical Livepatch service

This machine is not attached to a UA subscription.
See https://ubuntu.com/advantage
```
It could certainly be separated into its own plugin (I could work on writing that, too), but it just isn't clear exactly where the benefit would be by doing that (leaving only ...
```
        self.add_cmd_output([
            "ubuntu-security-status --thirdparty --unavailable",
            "hwe-support-status --verbose",
        ])
```
... on this 'ubuntu.py' plugin)

Reply:
If no actual benefit, I'm fine with keeping it as is. I think it was worth the exercise to think about it.
I would add the is_installed() method and also move ua-tool stuff inside the condition statement .